lib/pbench/test/unit/server/test_file_tree.py

```diff
@@ -190,6 +190,17 @@ def test_unpack(
     ):
         """Test to check the unpacking functionality of the FileTree"""
         source_tarball, source_md5, md5 = tarball
+        tarball_name = source_tarball.name.removesuffix(".tar.xz")
+
+        def mock_rmtree(path: pathlib.Path, ignore_errors=False):
+            unpack_dir = server_config.INCOMING / "ABC" / f"{tarball_name}.unpack"
+            assert path == unpack_dir
+
+        def mock_symlink(path: pathlib.Path, link: pathlib.Path):
+            results_dir = server_config.RESULTS / "ABC" / tarball_name
+            incoming_dir = server_config.INCOMING / "ABC" / tarball_name
+            assert path == results_dir
+            assert link == incoming_dir
 
         def mock_run(args, **kwargs):
             assert kwargs["capture_output"]
```

Comment:
I think it would be good to have the test fail if there is no call made to `rmtree()` (likewise for `symlink()`).  That is, we have these mocks which, _if they are called_, ensure that they are called correctly, but they produce no side-effects which the test scenario can check to ensure that they were actually called.  

When we were doing more "functional" unit tests, we would have checked the file system for the presence or absence of artifacts to confirm the correct behavior; I think we should add the equivalent to the test here.

The problem is that it's slightly tricky in Python for a mock to convey information to the containing code.  (There are lots of pitfalls which Python's binding rules make it easy to fall into.)  One way to do this is to have the mock set a variable which was initialized in the outer scope, but you'll need to declare the variable from the outer scope as `nonlocal`:
```
        rmtree_called = False

        def mock_rmtree(path: pathlib.Path, ignore_errors=False):
            nonlocal rmtree_called
            rmtree_called = True

            unpack_dir = server_config.INCOMING / "ABC" / f"{tarball_name}.unpack"
            assert path == unpack_dir

[...]

        assert rmtree_called
```
and do something similar for the `symlink_to()` mock.

That will work for tests with a small number of mock calls; however, in this test, instead of a collection of individual flags, you could also have the mocks append tokens to a single list.  So, the `mock_rmtree()` mock would append `"rmtree"` to the list, and the `mock_symlink()` would append `"simlink_to"`.   This would be nice because you could have `mock_run()` append `args[0]`, so you could tell the two calls apart.  And then, at the end of the test, you would have an assertion like
```
        assert calls == ["tar", "find", "symlink_to", "rmtree"]
```
So that you know both that all the mocks were called (the right number of times) and even that they were called in the right order.